packages/lib-react-components/src/ZooHeader/components/UserNavigation/UserNavigation.js

```diff
@@ -0,0 +1,123 @@
+import PropTypes from 'prop-types'
+import { Blank } from 'grommet-icons'
+import { useTranslation } from '../../../translations/i18n'
+
+import { faBell as fasBell } from '@fortawesome/free-solid-svg-icons/faBell'
+import { faBell as farBell } from '@fortawesome/free-regular-svg-icons/faBell'
+import { faEnvelope as fasEnvelope } from '@fortawesome/free-solid-svg-icons/faEnvelope'
+import { faEnvelope as farEnvelope } from '@fortawesome/free-regular-svg-icons/faEnvelope'
+import styled from 'styled-components'
+
+import NavButton from './components/NavButton/NavButton.js'
+import NavListItem from '../NavListItem'
+import UserMenu from '../UserMenu'
+import { getHost } from '../../helpers'
+
+const StyledBlank = styled(Blank)`
+  height: 1em;
+  vertical-align: -0.125em;
+`
+function FontAwesomeIcon({ color, icon, title }) {
+  const [width, height, aliases, unicode, path] = icon.icon
+  return (
+    <StyledBlank
+      role='img'
+      aria-label={title}
+      aria-hidden='false'
+      color={color}
+      viewBox={`0 0 ${width} ${height}`}
+    >
+      <path strokeWidth='1' fill='currentColor' d={path} />
+    </StyledBlank>
+  )
+}
+
+export default function UserNavigation({
+  isNarrow = false,
+  register,
+  signIn,
+  unreadMessages = 0,
+  unreadNotifications = 0,
+  signOut,
+  user
+}) {
+  const { t } = useTranslation()
+  const host = getHost()
+
+  const notificationLabelString = t(
+    'ZooHeader.SignedInUserNavigation.navListLabels.notifications',
+    { count: unreadNotifications }
+  )
+
+  const messagesLabelString = t(
+    'ZooHeader.SignedInUserNavigation.navListLabels.messages',
+    { count: unreadMessages }
+  )
+
+  const notificationLabel = isNarrow ? (
+    <FontAwesomeIcon
+      title={notificationLabelString}
+      icon={unreadNotifications ? fasBell : farBell}
+    />
+  ) : (
+    notificationLabelString
+  )
+
+  const messagesLabel = isNarrow ? (
+    <FontAwesomeIcon
+      title={messagesLabelString}
+      icon={unreadMessages ? fasEnvelope : farEnvelope}
+    />
+  ) : (
+    messagesLabelString
+  )
+
+  return (
+    <>
+      {Object.keys(user).length === 0 && (
+        <>
+          <NavButton
+            label={t('ZooHeader.SignedOutUserNavigation.signIn')}
+            onClick={signIn}
+          />
+          <NavButton
+            label={t('ZooHeader.SignedOutUserNavigation.register')}
+            onClick={register}
+          />
+        </>
+      )}
+      {Object.keys(user).length > 0 && signOut && (
```

Comment:
Is there ever a scenario in which neither line 77 nor 89 get evaluated? I'd normally expect an if-else combo so _something_ gets rendered... Mostly curious with this particular bit of code.

Reply:
In ZooHeader, `user`'s default definition is an empty object `{}`, and UserNavigation only renders after ZooHeader has mounted. In this case line 77 or line 89 will be true. Probably could be refactored with an if-else or some other combo for the same effect.